src/hooks/useCopyToClipboard.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect, useState } from 'react';
+
+const copyToClipboard = (value: string) => {
+  if (typeof window === 'undefined') {
+    return Promise.resolve(false);
+  }
+
+  return navigator.clipboard
+    .writeText(value)
+    .then(() => true)
+    .catch(() => false);
+};
+
+export const useCopyToClipboard = (): [
+  boolean,
+  (text: string) => Promise<boolean>
+] => {
+  const [copied, setCopied] = useState(false);
+
+  async function copyText(text: string) {
+    const isCopied = await copyToClipboard(text);
```

Comment:
This function doesn't need to be async.

```js
const copyText = (text: string) => copyToClipboard(text).then(setCopied)
```

I don't think the function itself should return the status, I mean, why then we have a state with the status?

Reply:
Yeah we can remove it